platform/linux-generic/pktio/dpdk.c

```diff
@@ -2090,23 +2129,22 @@ static int dpdk_send(pktio_entry_t *pktio_entry, int index,
 		_odp_pktio_tx_ts_set(pktio_entry);
 
 	if (_ODP_DPDK_ZERO_COPY) {
-		/* Free copied packets */
-		if (odp_unlikely(copy_count)) {
-			uint16_t freed = 0;
-
-			for (i = 0; i < mbufs && freed != copy_count; i++) {
-				odp_packet_t pkt = pkt_table[i];
-				odp_packet_hdr_t *pkt_hdr = packet_hdr(pkt);
-
-				if (pkt_hdr->seg_count > 1) {
-					if (odp_likely(i < tx_pkts))
-						odp_packet_free(pkt);
-					else
-						rte_pktmbuf_free(tx_mbufs[i]);
-					freed++;
-				}
+		for (i = 0; i < mbufs; i++) {
```

Comment:
This adds additional loop to the fast path, which could be avoided by returning the number of packets requiring postprocessing from `pkt_to_mbuf_zero()`.

Reply:
Should be more optimized in v2.